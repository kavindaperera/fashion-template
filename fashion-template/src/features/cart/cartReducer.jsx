import { GET_CART } from './cartConstants';
import { createReducer } from "../../app/common/util/reducerUtil";

const initialState = {
    cart: []
  }

export const getCart = (state, payload) => {
    return {...state, cart:payload.cart}
}

export default createReducer(initialState, {
    [GET_CART]: getCart
})
