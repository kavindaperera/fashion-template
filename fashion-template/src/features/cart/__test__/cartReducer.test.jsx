import * as types from '../cartConstants';
import reducer from '../cartReducer';
import { initialState  } from '../cartReducer';


const mockCart1 = [
    {
      "added": "2020-05-27",
      "item": "2ByX54HZo5CKxXsUbrIx",
      "price": 98,
      "quantity": 1,
      "subItem": 0
    },
    {
      "added": "2020-05-27",
      "item": "2ByX54HZo5CKxXsUbrIx",
      "price": 98,
      "quantity": 1,
      "subItem": 1
    }
]

const mockCart2 = []

describe('cart reducer', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toEqual(
        {cart: []}
      )
    })
    it('should handle GET_CART', () => {
        expect(
          reducer(initialState , {
            type: types.GET_CART,
            payload: {cart: mockCart1}
          })
        ).toEqual({
            cart : mockCart1
          })

      })

      it('should handle GET_CART snapshot 1', () => {
        expect(
          reducer(initialState , {
            type: types.GET_CART,
            payload: {cart: mockCart1}
          })
        ).toMatchSnapshot()

      })

      it('should handle GET_CART', () => {
        expect(
          reducer(initialState , {
            type: types.GET_CART,
            payload: {cart: mockCart2}
          })
        ).toEqual({
            cart : mockCart2
          })

      })

      it('should handle GET_CART snapshot 2', () => {
        expect(
          reducer(initialState , {
            type: types.GET_CART,
            payload: {cart: mockCart2}
          })
        ).toMatchSnapshot()

      })
})