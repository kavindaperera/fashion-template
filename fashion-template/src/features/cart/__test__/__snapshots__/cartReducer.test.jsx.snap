// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart reducer should handle GET_CART snapshot 1 1`] = `
Object {
  "cart": Array [
    Object {
      "added": "2020-05-27",
      "item": "2ByX54HZo5CKxXsUbrIx",
      "price": 98,
      "quantity": 1,
      "subItem": 0,
    },
    Object {
      "added": "2020-05-27",
      "item": "2ByX54HZo5CKxXsUbrIx",
      "price": 98,
      "quantity": 1,
      "subItem": 1,
    },
  ],
}
`;

exports[`cart reducer should handle GET_CART snapshot 2 1`] = `
Object {
  "cart": Array [],
}
`;
